plugins/catalog-react/src/hooks/useEntityListProvider.tsx

```diff
@@ -155,39 +164,70 @@ export const EntityListProvider = <EntityFilters extends DefaultEntityFilters>(
   // update of the URL or two catalog sidebar links with different catalog filters.
   const location = useLocation();
 
-  const enablePagination =
-    props.pagination === true || typeof props.pagination === 'object';
-
-  const limit =
-    props.pagination &&
-    typeof props.pagination === 'object' &&
-    typeof props.pagination.limit === 'number'
-      ? props.pagination.limit
-      : 20;
-
-  const { queryParameters, cursor: initialCursor } = useMemo(() => {
+  let paginationMode: PaginationMode =
+    typeof props.pagination === 'object'
+      ? props.pagination.mode ?? 'cursor'
+      : 'none';
+  if (props.pagination === true) {
+    paginationMode = 'cursor';
+  }
+
+  const {
+    queryParameters,
+    cursor: initialCursor,
+    offset: initialOffset,
+    limit: initialLimit,
+  } = useMemo(() => {
     const parsed = qs.parse(location.search, {
       ignoreQueryPrefix: true,
     });
+    let limit = 20;
+    if (typeof props.pagination === 'object') {
+      limit = props.pagination.limit ?? 20;
+    }
+
+    if (typeof parsed.limit === 'string') {
+      const queryLimit = Number.parseInt(parsed.limit, 10);
+      if (!isNaN(queryLimit)) {
+        limit = queryLimit;
+      }
+    }
+
+    const offset =
+      typeof parsed.offset === 'string' && paginationMode === 'offset'
+        ? Number.parseInt(parsed.offset, 10)
+        : undefined;
 
     return {
       queryParameters: (parsed.filters ?? {}) as Record<
```

Comment:
Do we want to exclude the limit and offset from this? Do their presence pollute something elsewhere? Maybe it's perfectly ok but the type does imply that it should only contain keys that match the installed filters

Reply:
I would say they allow users to share their specific view of the catalog with URL parameters so they are quite handy.